src/SINDYPI_fit.jl

```diff
@@ -0,0 +1,457 @@
+##FIT FUNCTION
+
+function fit!(X::AbstractArray, Y::AbstractArray, maxiter, lambda, Polyorder, Trigorder)

```

Comment:
The function does not mutate anything, so it should be named `fit(...)`. 
I suggested `fit!(...)`this based on the assumption you wanted to implement a sparse regression algorithm rather than a independent sindy function.